lldb/source/Plugins/TypeSystem/Swift/TypeSystemSwiftTypeRef.cpp

```diff
@@ -2065,13 +2065,30 @@ lldb::Encoding TypeSystemSwiftTypeRef::GetEncoding(opaque_compiler_type_t type,
 lldb::Format TypeSystemSwiftTypeRef::GetFormat(opaque_compiler_type_t type) {
   return m_swift_ast_context->GetFormat(ReconstructType(type));
 }
+
 uint32_t
 TypeSystemSwiftTypeRef::GetNumChildren(opaque_compiler_type_t type,
                                        bool omit_empty_base_classes,
                                        const ExecutionContext *exe_ctx) {
-  return m_swift_ast_context->GetNumChildren(ReconstructType(type),
-                                             omit_empty_base_classes, exe_ctx);
+  auto impl = [&]() -> uint32_t {
+    if (!exe_ctx)
+      return 0;
+
+    // TODO: which of these should be logged on failure?
```

Comment:
Since this function returns `uint32_t` and not an optional, I think some logging in is called for. But I'm not sure which of these should log, at least the call to `SwiftLanguageRuntime::GetNumChildren`, but maybe only that one.

Reply:
That's something we should just change upstream, whenever we find time for it. This interface predates `llvm::Optional<>`, but that shouldn't prevent us from improving it.

Reply:
I'll be sure to do that.